### ISSUE TITLE
Stabilize Experimental Rule: `list-style`

### ISSUE THREAD

**Issue by 9romise** [eslint-stylistic/eslint-stylistic#1030]
Initial PR: #895

- [x] Use in this repository
- [ ] Use in `@antfu/eslint-config` - https://github.com/antfu/eslint-config/pull/770

Any issues reported during stabilization will be addressed with high priority.

**Comment by VandinVanadium** [comment]
Maybe its manual page should be looked over, perhaps the text "Examples of incorrect code for this rule with the "always" 'option:', is meant to be applied to "never"?

[https://eslint.style/rules/list-style](https://eslint.style/rules/list-style)

**Comment by 9romise** [comment]
> Maybe its manual page should be looked over, perhaps the text "Examples of incorrect code for this rule with the "always" 'option:', is meant to be applied to "never"?
> 
> [eslint.style/rules/list-style](https://eslint.style/rules/list-style)

Thanks for pointing that out. I think the example itself is correct — the apparent spacing issue is caused by how `twoslash-eslint` handles empty characters during rendering.

**Comment by susnux** [comment]
We have found following edge case:

```js
// 🚧 Currently this is an edge case and isn't fixed properly...
const WEEKDAYS = ['Monday',
	'Tuesday',
	'Wednesday',
	'Thursday',
	'Friday',
	'Saturday',
	'Sunday',
]
```

The expected fix (default config) would be:
```js
// 🚧 Currently this is an edge case and isn't fixed properly...
const WEEKDAYS = [
	'Monday',
	'Tuesday',
	'Wednesday',
	'Thursday',
	'Friday',
	'Saturday',
	'Sunday',
]
```

But currently it is fixed to (also not the tabs are kept):
```js
const WEEKDAYS = ['Monday',	'Tuesday',	'Wednesday',	'Thursday',	'Friday',	'Saturday',	'Sunday']
```

**Comment by 9romise** [comment]
@susnux Sorry for the late replay.

The current rule behavior is designed to preserve the original formatting intent: if the first element is on the same line as the opening bracket, it will format as single-line; if there's a line break, it will format as multi-line. This allows developers to easily toggle between single-line and multi-line layouts.

For your specific case, you can configure the [`singleLine#maxItems`](https://eslint.style/rules/list-style#maxitems) option to control when arrays should wrap. Setting it to `1` would enforce multi-line formatting regardless of the original layout.

Regarding the tab preservation issue — this is already being tracked in #1062 and will be addressed separately.

**Comment by susnux** [comment]
@9romise thank you that solved our problem :) (ref #1062)

**Comment by susnux** [comment]
We found another issue with this rule, you can see it here:

https://stackblitz.com/edit/github-tztap4mh?file=README.md,src%2Findex.ts

<img width="1045" height="719" alt="Image" src="https://github.com/user-attachments/assets/41aba542-0605-4932-927b-c5248d9956ab" />

**Comment by jrmajor** [comment]
Currently, `array-bracket-spacing` and `object-curly-spacing` can handle spacing inside empty arrays/objects. If this rule is meant to replace them, it would be nice if it could handle that as well.

I've also come across an edge case:
```ts
function foo([x, y]: Coordinates) {
}
```
For this snippet, I get the following error:
```
Should not have space(s) between ':' and 'Coordinates'
```

Please let me know if I should open separate issue for this.

**Comment by bartschriever** [comment]
Any updates about when this rule will become non-experimental?
